Feature: 헥사고날 아키텍처 기반 CSP 자원 연동 구현 통합

hyobin-yang opened the pull request:
헥사고날 아키텍처 기반 CSP 자원 연동 구현 통합을 위한 pr입니다.

## 💡 점검사항 1: capability 패키지의 위치
- 현재 위치: com.agenticcp.core.domain.cloud.capability
- capability 패키지 내부 로직들은 CSP가 '정지' 기능을 지원하는가? 등을 확인하는 코어 로직 + CloudResourceUseCaseService 등의 코어 서비스가 호출하는 로직이므로 외부가 아니라 '도메인 코어'의 일부로 보고 service 패키지 안으로 이동시키는 것은 어떤지?
- 즉 `com.agenticcp.core.domain.cloud.capability` -> `com.agenticcp.core.domain.cloud.service.capability`

## 💡 점검사항 2: 자격증명의 위치
- 11.17 회의 때 논의 후 결정

## 💡 점검사항 3: 코드 통일
- 현재 아직 통일되지 않은 부분들 다수 존재(커스텀 에러, config 등)
- 각자 코드 우선 머지한 후 일괄적으로 통일
- 모든 헥사고날 로직은 이 `merge/hexagonal/aws`에 통합시킨 후 완료되면 develop 방향으로 머지

## 💡 점검사항 4: 주석 및 예외처리 적용 확인
- 헥사고날 관련한 Issue #144 과제는 헥사고날 관련 브랜치에서 수행

## 💡 점검사항 5: 부수적인 어댑터들
- adapter/outbound/TracingPortAdapter.java
- adapter/outbound/common/TracingPortAdapter.java
- adapter/outbound/audit/AuditEventPortAdapter.java
- adapter/outbound/common/AuditEventPortAdapter.java
- 위 어댑터들..

juanxiu committed:
build: aws sdk 의존성 추가

juanxiu committed:
feat: AwsConfig 추가

juanxiu committed:
feat: AwsVpcMapper 응답 모델 변환 메서드

juanxiu committed:
feat: Vpc 생성 요청 dto

juanxiu committed:
feat: VpcManagementPort

juanxiu committed:
feat: VpcPortRouter

juanxiu committed:
feat: VpcUseCaseService

juanxiu committed:
feat: AwsVpcManagementAdapter

juanxiu committed:
feat: VpcController

juanxiu committed:
test: VpcUseCaseServiceTest 테스트 성공

juanxiu committed:
feat: 어댑터에 오버라이드 메서드 추가

juanxiu committed:
feat: AuditLog 감사 어노테이션 적용

juanxiu committed:
feat: 수정, 삭제 기능 구현

juanxiu committed:
feat: aws region 설정 추가

moonhyeonmin committed:
feat: AWS SDK v2 의존성 및 EC2 기본 설정 추가

- pom.xml에 AWS SDK v2 의존성 추가 (ec2, aws-core, auth)
- @Value 어노테이션을 사용한 AwsConfig 구현
- application.yml과 application-local.yml에 AWS 설정 추가
- 환경변수 기반 설정 지원
- LocalStack 테스트용 엔드포인트 오버라이드 포함
- 조건부 로딩으로 EC2 클라이언트 빈 설정 (aws.enabled=true)

설정 내용:
- 리전 설정 (기본값: us-east-1)
- 자격증명 프로바이더 (정적 또는 기본)
- LocalStack 테스트용 엔드포인트 오버라이드
- 연결 및 읽기 타임아웃 설정

참고: AWS SDK 로드까지 EC2 클라이언트 생성 임시 비활성화

moonhyeonmin committed:
feat: Define EC2 domain models (Query, CreateRequest, UpdateRequest, DeleteRequest)

- Ec2Query.java - EC2 조회 조건 모델
- Ec2CreateRequest.java - EC2 생성 요청 모델
- Ec2UpdateRequest.java - EC2 수정 요청 모델
- Ec2DeleteRequest.java - EC2 삭제 요청 모델

moonhyeonmin committed:
feat: EC2 관리 포트 인터페이스 정의

- Ec2ManagementPort 인터페이스 구현
- EC2 인스턴스 전체 생명주기 관리 기능 정의
- 인스턴스 조회, 생성, 수정, 삭제, 생명주기 관리 메서드 포함
- 태그 관리 및 상태 확인 기능 추가
- 헥사고날 아키텍처의 핵심 포트(계약) 구현
- 상세한 JavaDoc으로 계약 명시 및 예외 상황 정의

주요 기능:
- 인스턴스 조회 (listInstances, getInstance)
- 생명주기 관리 (create, start, stop, reboot, terminate, delete)
- 인스턴스 수정 (updateInstance)
- 태그 관리 (addTags, removeTags, getTags)
- 상태 확인 (getInstanceStatus, waitForInstanceStatus)

moonhyeonmin committed:
feat: AWS EC2 매퍼 실제 SDK 통합 구현 완료

- 임시 Map 기반 구현을 실제 AWS SDK 클래스로 교체
- AWS Instance와 CloudResource 간 완전한 데이터 변환 구현
- 모든 EC2 작업 매핑 추가 (조회, 생성, 수정, 삭제)
- 인스턴스 타입별 CPU/메모리 매핑 및 메타데이터 추출 지원
- 데이터 변환을 위한 적절한 에러 처리 및 로깅 추가
- AwsConfig를 실제 Ec2Client.Builder 패턴 사용하도록 업데이트
- 매퍼 기능에 대한 단위 테스트 추가

변경사항:
- AwsEc2Mapper: Instance, DescribeInstancesRequest, RunInstancesRequest 등 완전한 AWS SDK 통합
- AwsConfig: 적절한 빌더 패턴으로 실제 Ec2Client 생성 활성화
- AwsEc2MapperSimpleTest: 기본 매퍼 기능 테스트

EC2 관리를 위한 핵사고날 아키텍처의 데이터 변환 계층을 완성합니다.

moonhyeonmin committed:
feat: AWS EC2 어댑터 구현 완료 - 모든 EC2 관리 기능 제공

- Ec2ManagementPort 인터페이스의 모든 메서드 구현
- AWS SDK와 도메인 계층을 연결하는 핵사고날 아키텍처 어댑터 완성
- 인스턴스 조회, 생성, 수정, 삭제, 생명주기 관리 기능 제공
- 태그 관리 및 상태 확인 기능 구현
- CloudErrorTranslator를 통한 에러 처리
- ProviderScoped 인터페이스 구현으로 AWS 제공자 식별

구현된 기능:
- listInstances: EC2 인스턴스 목록 조회 (페이징 지원)
- getInstance: 단일 인스턴스 조회
- createInstance: 새 인스턴스 생성
- startInstance/stopInstance/rebootInstance: 생명주기 관리
- terminateInstance/deleteInstance: 인스턴스 종료/삭제
- updateInstance: 인스턴스 정보 수정
- addTags/removeTags/getTags: 태그 관리
- getInstanceStatus/waitForInstanceStatus: 상태 확인 및 대기

핵사고날 아키텍처의 어댑터 계층이 완성되어 AWS EC2와의 통합이 가능합니다.

moonhyeonmin committed:
feat: EC2 포트 라우터 구현 완료 - 제공업체별 어댑터 동적 선택

- Ec2PortRouter 클래스 구현으로 제공업체별 EC2 어댑터 관리
- ProviderScoped 인터페이스를 통한 어댑터 자동 등록 및 식별
- 런타임에 제공업체 타입에 따라 적절한 어댑터 선택
- 지원되지 않는 제공업체에 대한 명확한 에러 처리
- 제공업체 지원 여부 확인 및 등록된 포트 정보 조회 기능

구현된 기능:
- ec2(ProviderType): 제공업체별 EC2 포트 반환
- getSupportedProviders(): 지원되는 제공업체 목록 조회
- isProviderSupported(): 제공업체 지원 여부 확인
- getPortCount(): 등록된 포트 개수 조회

핵사고날 아키텍처의 라우터 계층이 완성되어 다중 클라우드 환경에서
동적으로 적절한 어댑터를 선택할 수 있습니다.

moonhyeonmin committed:
feat: EC2 인스턴스 발견을 위한 유스케이스 서비스 구현

- 모든 CRUD 작업을 포함한 Ec2UseCaseService 완전 구현
- 모든 EC2 작업에 대한 포괄적인 감사 로깅 추가
- 인스턴스 생명주기 관리 구현 (시작, 중지, 재부팅, 종료)
- 태그 관리 기능 추가 (태그 추가, 제거, 조회)
- 상태 확인 및 대기 기능 포함
- Mock 검증을 포함한 포괄적인 단위 테스트 추가
- 적절한 관심사 분리와 함께 핵사고날 아키텍처 원칙 준수
- 적절한 트랜잭션 관리 및 로깅과 함께 서비스 패턴 적용

moonhyeonmin committed:
test: EC2 인스턴스 생성 기능에 대한 포괄적인 테스트 추가

- Ec2UseCaseServiceCreateTest 클래스 추가
- AwsEc2MapperCreateTest 클래스 추가
- 다양한 생성 시나리오 테스트 (기본, 전체옵션, 태그포함, 예외처리)
- AWS SDK RunInstancesRequest 변환 로직 검증
- Mock 검증을 통한 포트 호출 확인
- 감사 로그 기록 검증

moonhyeonmin committed:
test: EC2 인스턴스 생성 기능에 대한 포괄적인 테스트 추가

- Ec2UseCaseServiceCreateTest 클래스 추가
- AwsEc2MapperCreateTest 클래스 추가
- 다양한 생성 시나리오 테스트 (기본, 전체옵션, 태그포함, 예외처리)
- AWS SDK RunInstancesRequest 변환 로직 검증
- Mock 검증을 통한 포트 호출 확인
- 감사 로그 기록 검증

moonhyeonmin committed:
feat: EC2 REST API 컨트롤러 구현

- EC2 인스턴스 관리를 위한 포괄적인 REST API 엔드포인트 추가
- 모든 CRUD 작업 지원: 조회, 생성, 수정, 삭제
- 생명주기 관리 구현: 시작, 중지, 재부팅, 종료
- 태그 관리 추가: 태그 추가, 제거, 조회
- 상태 모니터링 포함: 상태 확인, 상태 대기
- 모든 엔드포인트에 대한 완전한 Swagger/OpenAPI 문서화
- 적절한 HTTP 메서드를 사용한 RESTful 설계 원칙 준수
- 포괄적인 로깅 및 에러 처리 추가
- 비즈니스 로직을 위한 Ec2UseCaseService와 통합
- 인스턴스 목록 조회를 위한 페이징 지원
- 성공 및 에러 응답을 적절히 처리

API 엔드포인트:
- GET    /api/v1/ec2/instances                    # 인스턴스 목록 조회
- GET    /api/v1/ec2/instances/{id}               # 인스턴스 조회
- POST   /api/v1/ec2/instances                    # 인스턴스 생성
- PUT    /api/v1/ec2/instances/{id}               # 인스턴스 수정
- DELETE /api/v1/ec2/instances/{id}               # 인스턴스 삭제
- POST   /api/v1/ec2/instances/{id}/start         # 인스턴스 시작
- POST   /api/v1/ec2/instances/{id}/stop          # 인스턴스 중지
- POST   /api/v1/ec2/instances/{id}/reboot        # 인스턴스 재부팅
- POST   /api/v1/ec2/instances/{id}/terminate     # 인스턴스 종료
- POST   /api/v1/ec2/instances/{id}/tags          # 태그 추가
- DELETE /api/v1/ec2/instances/{id}/tags          # 태그 제거
- GET    /api/v1/ec2/instances/{id}/tags          # 태그 조회
- GET    /api/v1/ec2/instances/{id}/status        # 상태 확인
- POST   /api/v1/ec2/instances/{id}/wait          # 상태 대기

이로써 EC2 관리를 위한 핵사고날 아키텍처 구현이 완료되어,
외부 클라이언트를 위한 완전한 REST API 인터페이스를 제공합니다.

moonhyeonmin committed:
test: EC2 계약 테스트 및 통합 테스트 추가

- 포트 인터페이스 계약 검증을 위한 Ec2ManagementContractTest 추가
- 어댑터 기본 기능 테스트를 위한 AwsEc2ManagementAdapterTest 추가
- REST API 엔드포인트 테스트를 위한 Ec2ControllerTest 추가
- EC2 관리 계층의 모든 테스트 커버리지 완성
- 포괄적인 Mock 테스트로 포트 인터페이스 계약 검증
- 어댑터 의존성 주입 및 제공업체 타입 식별 테스트
- 컨트롤러 HTTP 메서드 및 요청/응답 처리 테스트
- 적절한 에러 처리 및 엣지 케이스 커버리지 보장

테스트 커버리지:
- 포트 인터페이스 계약 테스트 (Ec2ManagementContractTest)
- 어댑터 통합 테스트 (AwsEc2ManagementAdapterTest)
- 컨트롤러 API 테스트 (Ec2ControllerTest)
- 기존 유스케이스 서비스 테스트 (Ec2UseCaseServiceTest*)
- 기존 매퍼 테스트 (AwsEc2MapperTest*)
- 기존 라우터 테스트 (Ec2PortRouterTest)

이로써 EC2 관리 시스템의 포괄적인 테스트 스위트가 완성되어,
모든 아키텍처 계층에서의 신뢰성과 유지보수성이 확보되었습니다.

moonhyeonmin committed:
fix: 테스트 환경에서 Ec2Client Bean 누락으로 인한 ApplicationContext 로딩 실패 해결

- MockAdaptersConfig에 Mock Ec2Client와 AwsEc2Mapper Bean 추가
- 테스트 환경에서 AWS SDK 의존성 없이도 정상 동작하도록 개선
- AgenticCpCoreApplicationTests.contextLoads 테스트 정상 통과 확인

Resolves: ApplicationContext 로딩 실패 문제

moonhyeonmin committed:
feat: 실제 구현 환경에 맞게 클라우드 어댑터들 구현

- AuditEventAdapter: 기존 감사 시스템과 연동하여 일관된 감사 로깅 제공
  * AuditLogger와 AuditEventDto 활용
  * AuditResourceType.CLOUD_PROVIDER 사용
  * 결과에 따른 자동 심각도 결정 및 속성 추출

- CredentialProviderAdapter: 암호화된 자격증명 관리 및 캐시 기능 구현
  * EncryptionService를 통한 암호화된 자격증명 처리
  * application.yml 설정 기반 AWS 자격증명 관리
  * 메모리 기반 캐시로 성능 최적화
  * 테넌트별 자격증명 및 캐시 무효화 기능

- TracingAdapter: MDC 기반 분산 추적 시스템 구현
  * ThreadLocal 기반 스팬 컨텍스트 관리
  * 계층적 부모-자식 스팬 관계 지원
  * MDC 연동으로 로그에 추적 정보 자동 포함
  * 설정 기반 추적 기능 활성화/비활성화

모든 어댑터가 프로젝트의 실제 구조와 패턴에 맞게 구현되어
향후 외부 시스템과의 연동 준비 완료

moonhyeonmin committed:
feat(ec2): 실제 AWS 연동 완료 및 EC2 API 안정화
AwsEc2Config: DefaultCredentialsProvider 적용(AWS CLI 자격증명 사용), 설정 단순화
application-local.yml/start-dev.sh: 환경변수 정리(AWS 키 제거), CLI 방식 권장
Ec2CreateRequest: @Jacksonized + @Min(1)로 minCount/maxCount 기본값·검증 보장
Ec2Controller: @Valid 적용으로 입력 검증 강화
AwsEc2Mapper: min/max 하드코딩 보정 제거, 순수 매핑 유지
에러 로깅 개선: AWS 원인 메시지 노출(InvalidAMIID.NotFound 등)로 디버깅 용이
EC2 삭제 엔드포인트 사용 가이드 반영(경로 변수 필수)
Refs: feature/#130

moonhyeonmin committed:
feat: CapabilityGuard 적용 및 EC2 capability 등록, 테스트 호환성 개선
Ec2CapabilityConfig: AWS|EC2|INSTANCE capability 등록(start/stop/terminate/tagging 지원)
Ec2UseCaseService:
CapabilityGuard 선검증 적용(start/stop/reboot/terminate/delete/tags)
reboot는 START/STOP 모두 지원 검증
테스트 환경 호환을 위해 CapabilityGuard 선택 주입(@Autowired(required=false)) 처리
전체 동작 수동 검증(엔드포인트 정상, 가드 동작 확인)

moonhyeonmin committed:
feat(capability, ec2): 생성/수정에 CapabilityGuard 적용 및 플래그 추가

- CspCapability에 supportsCreate, supportsUpdate 플래그 추가
- CapabilityGuard.Operation에 CREATE/UPDATE 추가 및 검증 로직 반영
- Ec2CapabilityConfig에서 EC2 INSTANCE에 create/update 지원 등록
- Ec2UseCaseService의 create/update에 ensureSupported 적용
- 린트/컴파일 이상 없음

Refs: #130

syeongkim committed:
feat: CloudAccount 엔터티 추가

syeongkim committed:
feat: CloudAccountCredential 엔터티 추가

syeongkim committed:
feat: CloudAccountRepository 추가

syeongkim committed:
feat: CloudAccountCredentialRepository 추가

syeongkim committed:
feat: AccountStatus enum 추가

syeongkim committed:
feat: AccountValidationRequest, AccountValidationResult DTO 추가

syeongkim committed:
feat: ConnectionTestResult DTO 추가

syeongkim committed:
feat: AccountValidationPort 추가

syeongkim committed:
feat: AccountSyncPort 추가

syeongkim committed:
feat: CredentialProviderPort에 자격증명 저장, 삭제 메서드 추가

syeongkim committed:
chore: AWS SDK 의존성 추가

syeongkim committed:
feat: AwsClientConfig 추가

syeongkim committed:
feat: RegisterCloudAccountRequest, UpdateCloudAccountRequest DTO 추가

syeongkim committed:
feat: CloudAccountDto 추가

syeongkim committed:
feat: CloudAccountMapper 추가

syeongkim committed:
feat: CloudErrorCode 추가

syeongkim committed:
feat: AwsCredentialManager 추가

syeongkim committed:
feat: CloudAccountDomainService 추가

syeongkim committed:
feat: CloudAccountUseCaseService 추가

syeongkim committed:
feat: AwsAccountValidationAdapter 추가

syeongkim committed:
feat: AuditEventPortAdapter 추가

syeongkim committed:
feat: AccountSyncPortRouter, AccountValidationPortRouter 추가

syeongkim committed:
feat: CredentialProviderPortAdapter 추가

syeongkim committed:
feat: TracingPortAdapter 추가

syeongkim committed:
feat: AuditResourceType enum에 CLOUD_ACCOUNT 추가

syeongkim committed:
feat: AwsAccountSyncAdapter 추가

syeongkim committed:
feat: CloudAccountController 추가

syeongkim committed:
feat: CloudErrorCode 추가

syeongkim committed:
test: CloudAccountDomainServiceTest 추가

syeongkim committed:
fix: changeAccountStatus 메서드에서 상태 enum으로 비교하도록 수정

syeongkim committed:
test: CloudAccountUseCaseServiceTest 추가

syeongkim committed:
feat: CloudAccountUseCaseService 메서드 추가

syeongkim committed:
test: AwsAccountValidationAdapterTest 추가

syeongkim committed:
test: AwsCredentialManagerTest 추가

syeongkim committed:
feat: MockAdaptersConfig 메서드 추가

syeongkim committed:
test: CloudAccountDomainServiceTest 중복 검증 로직 추가

hyobin-yang committed:
refactor: Object Storage 추상화를 위한 AWS 종속 네이밍 제거

- S3/Bucket → ObjectStorage/Container로 네이밍 변경
- 프로바이더 종속성 제거하여 헥사고날 아키텍처 원칙 준수

hyobin-yang committed:
test: AWS 종속 네이밍 제거에 다른 테스트 코드 리팩토링

hyobin-yang committed:
refactor(ObjectStorageController): RESTful API 원칙에 맞게 컨트롤러 리팩토링

juanxiu committed:
fix: VpcController 에 try, catch 제거

juanxiu committed:
docs: vpc 생성 관련 api spec 문서 작성

juanxiu committed:
feat: AwsVpcCapabilityConfig

juanxiu committed:
Merge branch 'merge/hexagonal/aws' into feature/#123

hyobin-yang committed:
refactor: 포트 라우터 계층의 패키지명 수정

- aws -> storage

juanxiu committed:
refactor: vpccontroller 폴더 구조 변경

juanxiu committed:
feat: VpcUseCaseService 에 Audit 어노테이션 제거

juanxiu committed:
Merge branch 'feature/#123' of github.com:CCCloudPlatform/AgenticCP-Core into feature/#123

syeongkim committed:
fix: AccountValidationPortRouter에서 프로바이더 타입 확인 후 어댑터 라우팅하도록 수정

syeongkim committed:
fix: CredentialProviderPortAdapter에서 프로바이더 타입 확인 후 어댑터 라우팅하도록 수정

syeongkim committed:
fix: command 객체 적용

syeongkim committed:
feat: CredentialProviderPort에 세션 자격증명 관리 기능 추가

syeongkim committed:
feat: 세션 자격증명 관리 기능 구현

syeongkim committed:
feat: 세션 관련 에러코드 추가

syeongkim committed:
feat: ResourceLifecyclePort 인터페이스 수정

syeongkim committed:
feat: AwsResourceLifecycleAdapter에

syeongkim committed:
feat: AwsSessionProvider 추가

syeongkim committed:
feat: AwsSessionCredential 추가

syeongkim committed:
feat: CloudSessionCredential 인터페이스 추가

syeongkim committed:
feat: CloudResourceUseCaseService에서 세션 획득 로직 추가

syeongkim committed:
feat: SessionCacheService 추가

syeongkim committed:
feat: MockAdaptersConfig에 AwsSessionCredential 추가

syeongkim committed:
test: CredentialProviderPortAdapterGetSessionTest 추가

syeongkim committed:
test: AwsSessionProviderTest 추가

syeongkim committed:
test: CloudSessionCredentialTest 추가

syeongkim committed:
test: SessionCacheServiceTest 추가

syeongkim committed:
fix: CloudErrorCode, CredentialErrorCode 분리

syeongkim committed:
fix: 자격증명 관련 에러 CredentialErrorCode로 수정

syeongkim committed:
Merge branch 'merge/hexagonal/aws' into feature/125

syeongkim committed:
fix: CloudErrorCode conflict 해결

syeongkim committed:
fix: 자격증명 에러 코드 수정

juanxiu committed:
Provider 타입 변경

juanxiu committed:
feat: command 객체 생성

juanxiu committed:
feat: VpcUseCaseService 에서 command 객체 변환 및 capabilityGuard 적용

moonhyeonmin committed:
feat(cloud-vm): EC2 명칭을 VM으로 통일 및 어댑터/설정/테스트 정리
도메인 모델 리네이밍: Ec2* → Vm* (Query/Create/Update/Delete)
애플리케이션 계층: Ec2UseCaseService/Controller → VmUseCaseService/Controller, 라우터 Ec2PortRouter → VmPortRouter
포트/어댑터: Ec2ManagementPort → VmManagementPort, AwsEc2* → AwsVm* (패키지 aws.vm로 이동)
설정: AwsEc2Config → AwsVmConfig
Capability: Ec2CapabilityConfig → VmCapabilityConfig (SERVICE_KEY ‘VM’)
API 경로: /api/v1/ec2 → /api/v1/vms
Resilience4j 키: cloud.aws.ec2.list → cloud.aws.vm.list
테스트 리팩터: 경로/클래스/문자열 VM 기준으로 수정

KwonSunJae committed:
Merge pull request #142 from CCCloudPlatform/feature/125

Feature/125: 테넌트별 클라우드 계정 연결 및 관리 기능

juanxiu committed:
refactor: aws 에 종속되지 않도록 파일 경로 이동

juanxiu committed:
refactor: VPC 관련 클래스 패키지 경로 정리 및 Command 패턴 적용

juanxiu committed:
feat(repository): CloudServiceRepository 추가 및 CloudRegionRepository 확장

juanxiu committed:
feat(vpc): VpcConstants 추가로 하드코딩 제거

juanxiu committed:
refactor(vpc): AwsVpcMapper에서 CloudResource 엔티티 조회 로직 추가

juanxiu committed:
refactor(vpc): Command 패턴 적용 및 VPC 로직 개선

juanxiu committed:
docs(vpc): API_VPC_SPEC.md에 Command 패턴 및 매핑 개선 내용 반영

juanxiu committed:
merge: merge/hexagonal/aws 브랜치 병합

juanxiu commented on src/main/java/com/agenticcp/core/domain/cloud/adapter/outbound/CredentialProviderPortAdapter.java:
```diff
@@ -0,0 +1,185 @@
+
+        switch (providerType) {
+            case AWS:
+                return awsCredentialManager.getCredentials(accountScope);
```
resolveCredentials() 메서드는 아래와 같은 문제가 있는 것 같습니닷 그런데 자격증명 관련 로직을 getSession() 메서드를 사용한다면 사실 이 메서드는 사용되지 않을 것이니 수정하지 않으셔도 문제는 안될 것 같습니다 확인만 부탁드려요~!!

```
문제: accountScope를 직접 credentialKey로 사용합니다.
accountScope = AWS Account ID (예: "123456789012")
credentialKey = UUID (예: "550e8400-e29b-41d4-a716-446655440000")
```

syeongkim replied:
오앗 계정 연결할 때 사용되는 부분이라 accountScope으로 credentialKey 조회해서 credentialKey로 자격증명 해결하는 방식으로 수정했습니다! 감사해용 😊

juanxiu commented on src/main/java/com/agenticcp/core/domain/cloud/adapter/outbound/CredentialProviderPortAdapter.java:
```diff
@@ -0,0 +1,185 @@
+    }
+    
+    @Override
+    public CloudSessionCredential getSession(String tenantKey, Long accountId, ProviderType providerType) {
```
getSeession() 메서드는 아래와 같은 흐름으로 잘 구현되어있는 것 같습니닷 전 VPC 관련 Usecaseservice 에서 이 메서드 사용해서 세션 획득하고 있습니다 


```
accountId (Long)로 CloudAccount 조회
CloudAccount.credential.credentialKey (UUID) 획득
credentialKey로 세션 발급
```

juanxiu reviewed (commented):
확인 부탁드립니다~!!

hyobin-yang commented on src/main/java/com/agenticcp/core/domain/cloud/adapter/outbound/aws/account/AwsCredentialManager.java:
```diff
@@ -0,0 +1,147 @@
+        private String region;
+    }
+}
+
```
요 service 패키지는 아키텍처의 내부인데 'AwsCredentialManager'이 aws라는 특정 외부 기술 관련 로직이라서 외부로 다시 이동을 시켜야 할 것 같아요~! 

> adapter/outbound/aws/AwsCredentialManager

요 위치는 어떤가요?

syeongkim replied:
오 좋습니당!

juanxiu committed:
feat(vpc): Command 객체에 CloudSessionCredential session 필드 추가

juanxiu committed:
feat(vpc): VpcUseCaseService에 세션 획득 및 accountScope 변환 로직 추가

juanxiu committed:
feat(vpc): AwsClientConfig에 세션 기반 Ec2Client 생성 메서드 추가

juanxiu committed:
refactor(vpc): AwsVpcManagementAdapter를 세션 기반으로 전환

juanxiu committed:
feat(vpc): VpcController에 @PreAuthorize 권한 체크 추가

hyobin-yang commented on src/main/java/com/agenticcp/core/domain/cloud/port/model/account/DeleteCredentialCommand.java:
```diff
@@ -0,0 +1,27 @@
+     */
+    String credentialKey;
+}
+
```
API의 입출력 명세(DTO/Command)는 port/model 안에 모여있게 하는 건 어떤가요?? command 자체를 레이어로 빼는 것보다 지금 기능별 분리 전략대로(storage 자원 관련한 dto, command는 port/model/storage/ 안에 모여있음)
`port/model/account` 로 account 패키지를 하나 만들어서 account 기능 안에 command를 모아두는 방식이요!

syeongkim replied:
덕분에 구조가 더 깔끔해졌습니닷 감사해용 😊

hyobin-yang commented on src/main/java/com/agenticcp/core/domain/cloud/adapter/outbound/aws/account/AwsSessionCredential.java:
```diff
@@ -0,0 +1,79 @@
+    public ProviderType getProviderType() {
+        return ProviderType.AWS;
+    }
+}
```
위치 이동~~!

moonhyeonmin committed:
fix: 반환값 수정 및 경로 수정

moonhyeonmin committed:
feat: CspCapability create, update 필드 제거

syeongkim committed:
fix: AzureSessionCredential, GcpSessionCredential import 및 관련 코드 삭제

syeongkim committed:
feat: AwsCredentialProviderPortAdapter에 자격증명 저장, 삭제, 세션 발급 기능 추가

syeongkim committed:
feat: CloudStubConfiguration에 자격증명 저장, 삭제, 세션 발급 기능 추가

syeongkim committed:
refactor: AwsCredentialManager 패키지 경로 변경

syeongkim committed:
refactor: CloudAccount 관련 DTO/Command 패키지 경로 변경

syeongkim committed:
fix: CredentialProviderPortAdapter.resolveCredentials() 메소드에서 credentialKey로 자격증명 해결하도록 수정

syeongkim committed:
fix: CloudAccount 엔터티에서 accountId -> accountScope 필드명 변경
- AWS의 accountId, Azure의 subscriptionId, GCP의 projectId를 accountScope로 추상화
- accountScope으로 credentialKey 조회 및 자격증명 해결

syeongkim committed:
refactor: AwsAccountCredential 관련 어댑터들 패키지 경로 이동

syeongkim committed:
refactor: AccountCredentialPortRouter 추가
- AccountCredentialManagementPort, AccountSyncPort, AccountValidationPort 구현체를 프로바이더 타입에 따라 라우팅합니다.
- 기존에 AccountSyncPortRouter, AccountValidationPortRouter, CredentialProviderPortAdapter에서 라우팅하던 기능을 AccountCredentialPortRouter로 통합합니다.

syeongkim committed:
feat: AWS 자격증명 관련 어댑터들에 getProviderType 메서드 추가
AwsCredentialProviderPortAdapter -> AwsAccountCredentialManagementAdapter 이름 변경

syeongkim committed:
refactor: 자격증명 관련 port 패키지 리팩토링

syeongkim committed:
refactor: CloudSessionCredential, AwsSessionCredential 패키지 경로 변경

syeongkim committed:
refactor: 자격증명 관련 service 패키지 경로 이동

syeongkim committed:
test: CredentialProviderPortAdapterGetSessionTest 삭제 AwsAccountCredentialManagementAdapterTest에 통합

syeongkim committed:
refactor: 자격증명 관련 테스트

syeongkim committed:
fix: 리팩토링 반영해서 테스트 코드 수정

syeongkim committed:
fix: 불필요한 테스트 코드 제거

syeongkim committed:
fix: 중복되는 TracingPortAdapter와 AuditEventPortAdapter 제거

syeongkim committed:
fix: 불필요한 테스트 코드 제거

syeongkim committed:
test: AwsAccountSyncAdapterTest 추가

syeongkim committed:
feat: accessKey, accountScope 마스킹 전략 추가

syeongkim committed:
feat: 자격증명 관련 로깅에 마스킹 적용

syeongkim committed:
feat: 자격증명 관련 Command/DTO에 마스킹 적용

syeongkim committed:
docs: 클라우드 도메인 자격증명 관리 가이드라인 추가

syeongkim committed:
feat: AccountValidationResult에 AccountScope 마스킹 적용

syeongkim committed:
test: masking service mock 추가

syeongkim committed:
fix: AccessKey, AccountScope 마스킹 전략 수정

syeongkim committed:
feat: CloudAccountController 감사 로깅 적용

syeongkim committed:
feat: 계정 상태 변경 후 캐시 무효화 로직 추가

syeongkim committed:
refactor: SessionCacheService -> SessionCachePort, RedisSessionCacheAdapter 리팩토링

syeongkim committed:
test: SessionCacheService 리팩토링에 따라 SessionCacheServiceTest 수정

moonhyeonmin committed:
refactor: reorganize vm service layer

moonhyeonmin committed:
refactor: VM 포트 구조와 커맨드 분리

moonhyeonmin committed:
refactor: AWS VM 어댑터 역할 분리

moonhyeonmin committed:
chore: capability flag cleanup

syeongkim committed:
docs: 클라우드 자격증명 관리 가이드라인 수정

KwonSunJae committed:
Merge pull request #156 from CCCloudPlatform/fix/125

Fix/125: 클라우드 자격증명 관리 시스템 개선

juanxiu committed:
Merge branch 'merge/hexagonal/aws' of github.com:CCCloudPlatform/AgenticCP-Core into feature/#123

juanxiu committed:
refactor: CloudErrorCode 에러 코드 추가

juanxiu committed:
refactor: AccountCredentialManagementPort 사용, accountScope 직접 사용

juanxiu committed:
docs: 자격증명 가이드라인 준수 확인 문서

juanxiu committed:
fix(vpc): VPC Command 클래스들의 CloudSessionCredential import 경로 수정

juanxiu committed:
fix(aws): AwsClientConfig의 세션 자격증명 import 경로 수정

juanxiu committed:
test(vpc): VpcUseCaseService 단위 테스트 추가

KwonSunJae committed:
Merge pull request #135 from CCCloudPlatform/feature/#123

[FEATURE]VPC 생성 기능 구현

hyobin-yang committed:
refactor(AwsS3BucketManagementAdapter): 구조 리팩토링 및 자격증명 추가

hyobin-yang committed:
feat(AwsS3ErrorTranslator): S3 예외 Translator 클래스 구현

hyobin-yang committed:
refactor(AwsS3BucketDiscoveryAdapter): 구조 리팩토링 및 자격증명 추가

hyobin-yang committed:
refactor(AwsClientConfig): AWS Client 생성 로직 리팩토링

hyobin-yang committed:
refactor: RESTFul한 통신 위한 dto, command 필드 리팩토링

hyobin-yang committed:
refactor(ObjectStorageController): RESTFul하게 컨트롤러 구조 리팩토링

hyobin-yang committed:
refactor: 현 구조에 따라 VPC 파일 위치 변경

hyobin-yang committed:
refactor: 헥사고날 에러코드의 종속성 제거 및 추상화

hyobin-yang committed:
feat(AwsErrorTranslator): AWS 공통 예외 변환기 구현

hyobin-yang committed:
refactor: 기타 리팩토링

hyobin-yang committed:
test: 테스트 코드 리팩토링

hyobin-yang committed:
chore: 누락된 주석 추가

moonhyeonmin committed:
Merge branch 'merge/hexagonal/aws' of https://github.com/CCCloudPlatform/AgenticCP-Core into feature/#130

moonhyeonmin committed:
feat(VmCommand): VM Command 객체에 세션 자격증명 필드 추가

- VmCreateCommand, VmUpdateCommand, VmDeleteCommand에 CloudSessionCredential session 필드 추가
- 멀티 테넌트 환경에서 자격증명 격리를 위한 구조 개선

moonhyeonmin committed:
feat(cloud): AwsClientConfig 리팩토링

- validateAndCastSession(): 세션 검증 및 AWS 세션 타입 캐스팅
- toSdkCredentials(): 도메인 세션 → AWS SDK 세션 변환
- resolveRegion(): 리전 기본값 처리 (파라미터 > 세션 > us-east-1)
- createEc2Client(CloudSessionCredential, String): 세션 기반 EC2 클라이언트 생성

moonhyeonmin committed:
feat(vm): AwsVmLifecycleAdapter 세션 기반 클라이언트 생성 적용

Management 작업(create, update, delete)에 세션 기반 EC2 클라이언트 생성 패턴 적용

- AwsClientConfig 주입하여 세션 기반 클라이언트 생성
- createInstance: command.getSession()으로 클라이언트 생성
- updateInstance: command.getSession()으로 클라이언트 생성
- deleteInstance: command.getSession()으로 클라이언트 생성
- try-finally 패턴으로 클라이언트 리소스 해제

moonhyeonmin committed:
feat(vm): AwsVmDiscoveryAdapter Execute Around 패턴 적용

Discovery 작업에 JIT(Just-In-Time) 세션 획득 패턴 적용

- 싱글톤 Ec2Client 제거
- AwsClientConfig, AccountCredentialManagementPort 주입
- executeWithEc2Client(): 세션 획득 → 클라이언트 생성 → 작업 실행 → 해제
- listInstances, getInstance, getInstanceStatus, waitForInstanceStatus에 패턴 적용

moonhyeonmin committed:
feat(vm): VmLifecyclePort 시그니처 변경 - 세션 파라미터 추가

모든 생명주기 작업에 세션 자격증명 명시적 전달

Port 인터페이스 변경:
- startInstance(String, CloudSessionCredential)
- stopInstance(String, CloudSessionCredential)
- rebootInstance(String, CloudSessionCredential)
- terminateInstance(String, CloudSessionCredential)

Adapter 변경:
- 싱글톤 Ec2Client 완전 제거
- 모든 메서드가 세션 기반 클라이언트 생성/해제

Service 변경:
- 세션 획득 후 Port 메서드에 명시적 전달

moonhyeonmin committed:
test(vm): 테스트 코드 업데이트 - Phase 6 시그니처 변경 반영

VmLifecyclePort 시그니처 변경 및 Adapter 생성자 변경에 맞춰 테스트 수정

VmUseCaseServiceLifecycleTest:
- verify() 호출에 CloudSessionCredential 파라미터 추가
- doThrow() 설정에 세션 파라미터 매처 추가
- getSession() mock 설정 추가

moonhyeonmin committed:
test: VmManagementContractTest 수정

moonhyeonmin committed:
refactor(vm): ObjectStorageUseCaseService 스타일로 리팩토링

- providerType, accountScope를 메서드 파라미터로 분리
- 감사 로그 및 try-catch 제거
- 로깅 패턴 단순화
- Request 객체에서 providerType/accountScope 제거

moonhyeonmin committed:
refactor(vm): VmCreateRequest/Command를 CSP 중립 필드로 리팩토링

- VmCreateRequest, VmCreateCommand의 AWS 특화 필드를 CSP 중립 필드로 변경
  - imageId → image
  - instanceType → instanceSize
  - keyName → sshKey
  - securityGroupId → networkSecurityId
  - zone 필드 추가

- AwsVmMapper에 CSP 특화 변환 로직 추가
  - 도메인 모델(CSP 중립) → AWS SDK 요청(CSP 특화) 변환
  - zone → placement.availabilityZone 매핑 추가

moonhyeonmin committed:
refactor(vm): API Endpoint 통일

- URL 패턴 변경: /api/v1/vms → /api/v1/cloud/providers/{provider}/accounts/{accountScope}/vms/instances
- @RequestParam → @PathVariable로 providerType, accountScope 전달 방식 변경
- VmCreateRequest, VmDeleteRequest, VmUpdateRequest에 providerType, accountScope 필드 추가
- @Value → @Data 변경 (setter 지원)
- VmUseCaseService: Request 객체에서 providerType, accountScope 추출
- 테스트 코드 업데이트

moonhyeonmin committed:
refactor(vm): 세션 관리 패턴 통일 - Discovery/Tagging에도 getSession() 적용

- VmDiscoveryPort: 모든 메서드에 CloudSessionCredential 세션 파라미터 추가
- VmTaggingPort: 모든 메서드에 CloudSessionCredential 세션 파라미터 추가
- VmUseCaseService: acquireSession() → getSession()으로 변경, 모든 작업에서 세션 전달
- AwsVmDiscoveryAdapter: 세션 파라미터로 클라이언트 생성
- AwsVmTaggingAdapter: 세션 파라미터로 클라이언트 생성
- VmController: Discovery/Tagging 호출에 accountScope 전달

JIT 세션 관리 패턴 준수:
- Service 레벨에서 getSession()으로 세션 획득
- 획득한 세션을 Port/Adapter에 전달
- Redis 캐싱을 통한 세션 재사용

KwonSunJae committed:
Merge pull request #160 from CCCloudPlatform/refactor/hexagonal/s3

Refactor: Object Storage 관리 로직에 자격증명 추가 및 코드 리팩토링

hyobin-yang committed:
Merge branch 'merge/hexagonal/aws' into feature/#130

hyobin-yang committed:
Merge pull request #139 from CCCloudPlatform/feature/#130

[FEATURE] VM 인스턴스 연동 구현

hyobin-yang committed:
test: VM 테스트 오류 수정 및 import 경로 업데이트

hyobin-yang committed:
refactor: DTO와 Command 패키지 분리

- DTO는 웹 계층(dto/), Command는 도메인 계층(port/model/)으로 분리
- 계층 간 책임 명확화 및 의존성 방향 개선

hyobin-yang committed:
refactor: aws capability config 통합
- 기존에 aws 자원별로 분산되어있던 capability config 통합

hyobin-yang committed:
refactor: aws config 클래스 자원별 분리
- 상이한 자원별 config 클래스 분리

hyobin-yang committed:
refactor: redis 캐싱 로직 헥사고날로부터 분리
- 기존 port/adapter 패턴으로부터 분리
- 기존 service 명칭을 repository로 변경

hyobin-yang committed:
refactor: dto 경로 이동에 따른 import 업데이트

hyobin-yang committed:
test: 구조 변경에 따른 테스트 코드 리팩토링

hyobin-yang committed:
refactor: 사용하지 않는 어댑터 클래스 삭제

hyobin-yang committed:
Merge pull request #168 from CCCloudPlatform/merge/hexagonal/Integration

[REFACTOR] CSP 헥사고날 아키텍처 최종 통합 리팩토링

hyobin-yang committed:
Merge branch 'develop' of https://github.com/CCCloudPlatform/AgenticCP-Core into merge/hexagonal/aws

hyobin-yang committed:
Merge branch 'merge/hexagonal/aws' of https://github.com/CCCloudPlatform/AgenticCP-Core into merge/hexagonal/aws

moonhyeonmin committed:
feat: CloudResourceRepository 멀티 클라우드 자원 관리 메서드 추가

추가된 기능:
- CSP 독립적인 자원 조회 (프로바이더 타입별, 리소스 타입별)
- 생명주기 상태 조회 및 업데이트 (@Modifying)
- 소프트 삭제 처리
- 동기화 시간 업데이트
- 테넌트/프로바이더별 통계 조회

moonhyeonmin committed:
feat: VmUseCaseService DB 저장 로직 구현

변경 사항:
- createInstance: CSP 생성 후 CloudResource 신규 저장
- startInstance: lifecycleState → RUNNING 업데이트
- stopInstance: lifecycleState → STOPPED 업데이트
- rebootInstance: lifecycleState → RUNNING 유지
- terminateInstance: lifecycleState → TERMINATED 업데이트
- deleteInstance: 소프트 삭제 처리

추가된 의존성:
- CloudResourceRepository, CloudProviderRepository
- CloudServiceRepository, TenantRepository, ObjectMapper

moonhyeonmin committed:
feat: ObjectStorageUseCaseService DB 저장 로직 구현

변경 사항:
- createContainer: CSP 생성 후 CloudResource 신규 저장 (BUCKET 타입)
- deleteContainer: CSP 삭제 후 소프트 삭제 처리
- forceDeleteContainer: CSP 강제 삭제 후 소프트 삭제 처리

추가된 의존성:
- CloudResourceRepository, CloudProviderRepository
- CloudServiceRepository, TenantRepository, ObjectMapper

moonhyeonmin committed:
feat: VpcUseCaseService DB 저장 로직 구현

변경 사항:
- createVpc: CSP 생성 후 CloudResource 신규 저장 (NETWORK 타입)
- deleteVpc: CSP 삭제 후 소프트 삭제 처리

추가된 의존성:
- CloudResourceRepository, CloudProviderRepository
- CloudServiceRepository, TenantRepository, ObjectMapper

moonhyeonmin committed:
test: 멀티 클라우드 자원 DB 동기화 단위 테스트 추가 및 컴파일 오류 수정

추가된 테스트:
- VmUseCaseServiceDbSyncTest: VM 생성/시작/중지/종료/삭제 DB 동기화
- ObjectStorageUseCaseServiceDbSyncTest: 컨테이너 생성/삭제 DB 동기화
- VpcUseCaseServiceDbSyncTest: VPC 생성/삭제 DB 동기화

syeongkim committed:
fix: AuditService, AwsCredntialManager 트랜잭션 설정 수정

KwonSunJae commented:
@syeongkim 
<img width="1228" height="927" alt="스크린샷 2025-12-13 오후 12 08 32" src="https://github.com/user-attachments/assets/412cae73-b7db-460c-a6ff-0445a2a056e7" />
해결필용

moonhyeonmin committed:
Merge branch 'merge/hexagonal/aws' of https://github.com/CCCloudPlatform/AgenticCP-Core into feature/#169

hyobin-yang committed:
refactor: NoOpCloudPortsConfig에서 존재하지 않는 클래스 참조 수정

hyobin-yang commented:
> @syeongkim <img alt="스크린샷 2025-12-13 오후 12 08 32" width="1228" height="927" src="https://private-user-images.githubusercontent.com/32411719/526140606-412cae73-b7db-460c-a6ff-0445a2a056e7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU1OTg2MDUsIm5iZiI6MTc2NTU5ODMwNSwicGF0aCI6Ii8zMjQxMTcxOS81MjYxNDA2MDYtNDEyY2FlNzMtYjdkYi00NjBjLWE2ZmYtMDQ0NWEyYTA1NmU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjEzVDAzNTgyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0YjM4OTFkOGFjNDY3MzQyZjdmMTg4MTRhMThlNzJhYWEzNTJmODI1MWQ1ZjczODk5ZDA3YjBlMTU4MzM5ZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CHo-Tm-C_eIAsBkClZOq6E9i4B7rqLcwpdeWvza0BCM"> 해결필용

해결완

hyobin-yang committed:
fix: AwsCredentials 필드 마스킹 제거

hyobin-yang committed:
refactor: 마스킹 처리 시 원본 객체 변경 없도록 리팩토링

hyobin-yang committed:
test: 마스킹 처리 관련 테스트 리팩토링

syeongkim committed:
fix: credentialKey 조회 쿼리 수정

moonhyeonmin committed:
Merge branch 'merge/hexagonal/aws' of https://github.com/CCCloudPlatform/AgenticCP-Core into feature/#169

moonhyeonmin committed:
fix: Object Storage API 권한을 ADMIN에서 SUPER_ADMIN으로 변경

- ObjectStorageController의 모든 @PreAuthorize 어노테이션에서
  hasRole('ADMIN')을 hasRole('SUPER_ADMIN')으로 변경
- CREATE, UPDATE, READ, DELETE, HEAD 엔드포인트 모두 적용
- S3 버킷 생성/관리 권한을 더 엄격하게 제한

moonhyeonmin committed:
fix: Object Storage Capability 체크 로직을 BUCKET 리소스 타입으로 수정

- RESOURCE_TYPE 상수를 "BUCKET"으로 정의
- Capability 체크 시 "OBJECT_STORAGE"/"CONTAINER" 대신
  getServiceKeyForProvider()와 "BUCKET" 사용
- createContainer, updateContainer, deleteContainer, forceDeleteContainer
  모든 메서드에 적용
- AwsCapabilityConfig에 등록된 "AWS|S3|BUCKET" Capability와 일치하도록 수정

moonhyeonmin committed:
feat: AwsS3BucketManagementAdapter에 ProviderScoped 인터페이스 구현

- ProviderScoped 인터페이스를 구현하여 ObjectStoragePortRouter에서
  AWS 프로바이더로 어댑터를 식별할 수 있도록 함
- getProviderType() 메서드를 추가하여 CloudProvider.ProviderType.AWS 반환
- ObjectStoragePortRouter가 ProviderScoped를 구현한 어댑터만 필터링하여
  등록하므로, 이 변경으로 AWS S3 어댑터가 정상적으로 라우팅됨

moonhyeonmin committed:
fix: VpcCreateRequest에 @Jacksonized 어노테이션 추가

- Jackson이 @Value와 @Builder를 사용한 DTO를 역직렬화할 수 있도록
  @Jacksonized 어노테이션 추가
- VPC 생성 API 호출 시 HttpMessageConversionException 해결

moonhyeonmin committed:
fix: VPC API 권한을 ADMIN에서 SUPER_ADMIN으로 변경

- VpcController의 모든 @PreAuthorize 어노테이션에서
  hasRole('ADMIN')을 hasRole('SUPER_ADMIN')으로 변경
- CREATE, READ, UPDATE, DELETE 엔드포인트 모두 적용
- VPC 생성/관리 권한을 더 엄격하게 제한

moonhyeonmin committed:
refactor: VPC 서비스 키 로직 개선 및 멀티 클라우드 지원 준비

- getServiceKeyForProvider의 AWS 케이스를 "VPC"에서 "EC2"로 수정
  (VPC는 EC2 서비스에 속함)
- saveCloudResource에서 하드코딩된 VpcConstants.SERVICE_KEY 대신
  getServiceKeyForProvider(providerType)를 사용하도록 변경
- cloudService null 체크를 예외 처리로 변경하여 명확한 에러 메시지 제공
- saveCloudResource를 @Transactional(propagation = Propagation.REQUIRES_NEW)로
  분리하여 DB 저장 실패가 메인 트랜잭션에 영향을 주지 않도록 개선
- 향후 Azure, GCP 지원 시 코드 수정 없이 확장 가능하도록 구조 개선

moonhyeonmin committed:
refactor: CloudResource 생성 로직을 엔티티 Factory Method로 이동

moonhyeonmin committed:
refactor: JPA AttributeConverter 도입으로 JSON 직렬화 로직 제거

moonhyeonmin committed:
refactor: Repository 의존성 과다 해결 (Facade/Helper 도입)

moonhyeonmin committed:
fix: Ghost Resource 방지를 위한 보상 트랜잭션 구현

moonhyeonmin committed:
fix: VM 리소스 생성 API 응답에 리소스 ID 포함

moonhyeonmin committed:
fix: VPC 생성 시 vpcName이 resourceName으로 올바르게 설정되도록 수정

moonhyeonmin committed:
refactor: VPC 컨트롤러 API 패턴을 VM/ObjectStorage와 일관성 있게 수정

moonhyeonmin committed:
refactor: CloudResource 등록 로직을 통합 메서드로 리팩토링

moonhyeonmin committed:
refactor: CloudResource 등록 로직 통합 및 deprecated 메서드 제거

moonhyeonmin committed:
Merge pull request #171 from CCCloudPlatform/feature/#169

Feature/#169: 클라우드 리소스 변경 내역 DB 저장 로직 구현 및 트러블슈팅
